primo/utils/solvers.py

```diff
@@ -65,8 +64,28 @@ def get_solver(
         If an unrecognized solver is provided as an input.
         Supported solvers include glpk, gurobi, gurobi_persistent, highs, and scip
     """
-    if solver == "highs":
-        sol_obj = Highs()
+    if solver_options is None:
+        solver_options = {}
+
+    if solver is None:
+        # Auto-detect solver in order of priority
+        for solver_name in (
+            "gurobi_persistent",
+            "gurobi",
+            "scip",
+            "glpk",
+            "appsi_highs",
+        ):
+            if SolverFactory(solver_name).available(exception_flag=False):
+                LOGGER.warning(
+                    f"Optimization solver is not specified. "
+                    f"Using {solver_name} as the optimization solver."
+                )
+                solver = solver_name
+                break
```

Comment:
What if they have none of the listed solvers installed? Should we communicate it to them?